src/services/services.ts

```diff
@@ -473,7 +488,16 @@ namespace ts {
 
         getDocumentationComment(): SymbolDisplayPart[] {
             if (this.documentationComment === undefined) {
-                this.documentationComment = this.declaration ? JsDoc.getJsDocCommentsFromDeclarations([this.declaration]) : [];
+                if (this.declaration) {
+                    this.documentationComment = JsDoc.getJsDocCommentsFromDeclarations([this.declaration]);
+
+                    if (this.documentationComment.length === 0) {

```

Comment:
Great point!  I agree.  This would make TypeScript's handling of `@inheritDoc` different than vanilla JSDoc's though, which ignores documentation on anything with `@inheritDoc`, e.g.:

```javascript
/** @class */
class Foo {
    /** Foo docs. */
    test() {
    }
}

/**
 * @class
 * @extends Foo
 */
class Bar {
    /**
     * Bar docs.
     * @inheritDoc
     */
    test() {
    }
}
```

results in JSDoc for `Bar#test` of "Foo docs" only.

That said, I think it's helpful for TypeScript users and I'd love to have it.  As long as you're okay with differing from upstream here, I'm 100% fine with it 👍

Reply:
You're right, http://usejsdoc.org/tags-inheritdoc.html says:
> The @inheritdoc tag indicates that a symbol should inherit its documentation from its parent class. Any other tags that you include in the JSDoc comment will be ignored.
> This tag is provided for compatibility with Closure Compiler. By default, if you do not add a JSDoc comment to a symbol, the symbol will inherit documentation from its parent.

This makes it a rather strange tag -- one whose sole purpose is to make all documentation on the current node ignored! Presumably people who don't want to see that comment in quick info should just make it a regular comment and not a doc comment. I think it would be a more useful tag if it meant to *add* docs from a parent in addition to anything on the child.

Reply:
Makes perfect sense to me.  I've updated the PR to support that :)